src/mds/Locker.cc

```diff
@@ -5399,6 +5584,12 @@ void Locker::local_xlock_finish(const MutationImpl::lock_iterator& it, MutationI
   lock->put_xlock();
   mut->locks.erase(it);
 
+  if (lock->get_type() == CEPH_LOCK_IQUIESCE) {
+    auto in = static_cast<CInode*>(lock->get_parent());
+    // reevaluate everything related related to caps
+    eval(in, CEPH_LOCK_IFILE | CEPH_LOCK_IAUTH | CEPH_LOCK_ILINK | CEPH_LOCK_IXATTR, true);
```

Comment:
@batrick, maybe we should schedule this `eval` asynchronously? The reason would be to give potentially pending ops a chance to take their locks quickly and avoid the delay of revoking the write caps from the client again after this `eval` grants them.